### DISCUSSION TITLE
how to estimate fire position in time

### DISCUSSION THREAD

**Discussion by bjmvandongen** [firemodels/fds#13182] [General]
Hi all,

I am simulating an experiment where a bed of excelsior wood is set on fire on a sloped bench of 20 and 30 degrees (laboratory scale: width=3 m, length is 7 m).  I used the following line to get the fire position in time as output:

&DEVC ID='x', QUANTITY='TEMPERATURE', SPATIAL_STATISTIC='MAXLOC X', XB=0.1,7.0,-0.075,0.075,0,0.14 /

However, as can be seen in the figure below, it turns out that this definition for the fire front is not appropriate since the highest gas temperature is not at the front, but a bit behind (as the rate of spread graph suggests). The rate of spread graph shows a slower fire propagation than smokesview. 

![firefront](https://github.com/user-attachments/assets/07fecb46-a88b-4092-861f-30ddeca86365)

I was thinking that I can still get the fire front position at different time points when looking at the visualization in smokesview and at slices that indicate the temperature and oxygen mass fraction. However, I have to do that manually. 

Does someone know a better way to get the fire front position in time as output? Suggestions are very much welcome.

**Comment by mcgratta** [comment]
You could create a line of `POINTS` in the x direction near the surface along the centerline. Set `TIME_HISTORY=T` and there will be a number of columns in the _devc.csv file with the temperatures. When the temperature exceeds some chosen value, you can assume that the front has arrived. There may even be a clever way to do this within FDS using control functions.

**Reply by bjmvandongen** [reply]
Thank you, that is a helpful suggestion. I will try that.
